lib/git-metadata-preprocessor/extension.rb

```diff
@@ -0,0 +1,43 @@
+require 'asciidoctor/extensions' unless RUBY_ENGINE == 'opal'
+
+require 'rugged'
+
+class GitMetadataPreprocessor < Asciidoctor::Extensions::Preprocessor
+  def process document, reader
+
+    begin
+      repo = Rugged::Repository.discover(".")
+    rescue
+      $stderr.puts('Failed to find repository, git-metadata extension terminating')
+      exit
+    end
+
+    head = repo.head
+
+    document.attributes['git-metadata-sha'] = head.target_id
+    document.attributes['git-metadata-sha-short'] = head.target_id[0,8]
+    document.attributes['git-metadata-author-name'] = head.target.author[:name]
+    document.attributes['git-metadata-author-email'] = head.target.author[:email]
+    document.attributes['git-metadata-date-mdy'] = head.target.time.strftime("%m-%d-%y")
```

Comment:
For consistency with Asciidoctor, I will keep the date as `.strftime '%Y-%m-%d`: 
https://github.com/asciidoctor/asciidoctor/blob/d6608c90384ce425df68972c423ed0780371a9f5/lib/asciidoctor.rb#L1315

Reply:
Sounds fair to me.